tools/walletextension/api/routes.go

```diff
@@ -490,3 +494,84 @@ func versionRequestHandler(walletExt *walletextension.WalletExtension, userConn
 		walletExt.Logger().Error("error writing success response", log.ErrKey, err)
 	}
 }
+
+// ethRequestHandler parses the user eth request, passes it on to the WE to proxy it and processes the response
+func getMessageRequestHandler(walletExt *walletextension.WalletExtension, conn userconn.UserConn) {
+	// read the request
+	body, err := conn.ReadRequest()
+	if err != nil {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("error reading request: %w", err))
+		return
+	}
+
+	// read body of the request
+	var reqJSONMap map[string]interface{}
+	err = json.Unmarshal(body, &reqJSONMap)
+	if err != nil {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("could not unmarshal address request - %w", err))
```

Comment:
is this error correct?
What is the format  of the body?

Reply:
fixed error message. In the body we have "encryptionToken" and optionally "formats".